src/components/CommentAdder.jsx

```diff
@@ -0,0 +1,60 @@
+import { useState, useContext } from "react";
+import { UserContext } from "./User";
+import { postArticleComment } from "../../utils/api";
+
+export default function CommentAdder({ article_id, setCurrentComments }) {
+  const [newComment, setNewComment] = useState("");
+  const [commentPosted, setCommentPosted] = useState(false);
+  const [error, setError] = useState(null);
+  const { user } = useContext(UserContext);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    if (newComment.length < 5) {
+      setNewComment("");
+      return setError({ message: "Comment is too short!" });
+    } else {
+      setError(null);
+    }
+
+    const commentData = {
+      username: user.username,
+      body: newComment,
+    };
+
+    postArticleComment(article_id, commentData)
+      .then((comment) => {
+        setNewComment("");
+        setCommentPosted(false);
+        setCurrentComments((currComments) => [comment, ...currComments]);
+        setCommentPosted(true);
+        setError(null);
```

Comment:
There are a lot of set functions being invoked here, these are not guaranteed to happen in this order as they are all asynchronous functions.